web3/_utils/module.py

```diff
@@ -18,11 +18,13 @@
 
 def attach_modules(
     parent_module: Union["Web3", "Module"],
-    module_definitions: Dict[str, Sequence[Any]],
+    module_definitions: Dict[str, Any],
```

Comment:
The recursive nature of this method makes it really hard to add any useful typing here so this has to be `Any` unfortunately. But I am new to typing in python, maybe there's a way to make this stricter 🤔 . At least all the uses of this method are able to be a tad more strict.

Reply:
I found this documentation: https://mypy.readthedocs.io/en/stable/protocols.html#recursive-protocols, but I would have to play with it some to figure out how exactly we'd use it. If digging into that seems out of the scope of this PR, I'm fine leaving this as-is.